### PR TITLE
[Enhance] Quality Inspection Template

### LINKED ISSUE #2273: Quality Inspection [redesign]
- [x] Make Quality Inspection Template (Parameter, Acceptance Criteria)
- [x] In Item, select Quality Inspection Template (option to make applicable at time of Sale, Purchase)
- [x] In BOM, select Quality Inspection Template
- [ ] In stock transactions, if Quality Inspection is required. Add a HTML label + button "Inspection Required"
- [x] Do not allow submit if Quality Inspection is required and not added (via Stock Ledger Entry)
- [x] Applicable in Stock Entry, Delivery Note, Sales Invoice, Purchase Receipt

### PR DESCRIPTION
**Pull request by rohitwaghchaure**
- Added Inspection Required in BOM, Stock Entry
-  Added new doctype quality inspection template
- Removed quality inspection child table from item master and added link to select quality inspection template
- Added quality inspection template in quality inspection.

Fixed https://github.com/frappe/erpnext/issues/2273

### CODE REVIEW

**Comment by nabinhait** [comment]
@rohitwaghchaure Travis is failing

**Comment by rohitwaghchaure** [comment]
Fixed!

**Comment by nabinhait** [comment]
Fix the patch as per our discussion.

**Comment by rohitwaghchaure** [comment]
Fixed!

File: erpnext/patches/v11_0/make_quality_inspection_template.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		qc_doc = frappe.new_doc("Quality Inspection Template")
+		qc_doc.quality_inspection_template_name = data.parent
+		qc_doc.flags.ignore_mandatory = True
+		qc_doc.save(ignore_permissions=True)
+
+		item_doc = frappe.get_doc('Item', data.parent)
+		item_doc.db_set('quality_inspection_template', qc_doc.name)
+
+	frappe.db.sql(""" update `tabItem Quality Inspection Parameter`
+		set parentfield = 'item_quality_inspection_parameter', parenttype = 'Quality Inspection Template'
```
Comment (nabinhait) [inline erpnext/patches/v11_0/make_quality_inspection_template.py]: set parent as well (most important!!!)

File: erpnext/patches/v11_0/make_quality_inspection_template.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+	for data in frappe.get_all('Item Quality Inspection Parameter',
+		fields = ["distinct parent"], filters = {'parenttype': 'Item'}):
+		qc_doc = frappe.new_doc("Quality Inspection Template")
+		qc_doc.quality_inspection_template_name = data.parent
+		qc_doc.flags.ignore_mandatory = True
+		qc_doc.save(ignore_permissions=True)
+
+		item_doc = frappe.get_doc('Item', data.parent)
+		item_doc.db_set('quality_inspection_template', qc_doc.name)
```
Comment (nabinhait) [inline erpnext/patches/v11_0/make_quality_inspection_template.py]: no need for a `get_doc`, use `set_value`

File: erpnext/patches/v11_0/make_quality_inspection_template.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import frappe
+
+def execute():
+	frappe.reload_doc('stock', 'doctype', 'item')
+	frappe.reload_doc('stock', 'doctype', 'quality_inspection_template')
+
+	for data in frappe.get_all('Item Quality Inspection Parameter',
+		fields = ["distinct parent"], filters = {'parenttype': 'Item'}):
+		qc_doc = frappe.new_doc("Quality Inspection Template")
+		qc_doc.quality_inspection_template_name = data.parent
```
Comment (nabinhait) [inline erpnext/patches/v11_0/make_quality_inspection_template.py]: Instead of only item code, name can be like `"QIT/" + item_code`

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/patches/v11_0/make_quality_inspection_template.py (hunk 3 above)
Comment (rohitwaghchaure) [reply erpnext/patches/v11_0/make_quality_inspection_template.py]: Fixed!

**Pull request merged**